src/js/crypto.cpp

```diff
@@ -660,6 +660,130 @@ namespace ccf::js
     }
   }
 
+  static JSValue js_unwrap_key(
+    JSContext* ctx, JSValueConst, int argc, JSValueConst* argv)
+  {
+    if (argc != 3)
+      return JS_ThrowTypeError(
+        ctx, "Passed %d arguments, but expected 3", argc);
+
+    // API loosely modeled after
+    // https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/wrapKey.
+
+    size_t key_size;
+    uint8_t* key = JS_GetArrayBuffer(ctx, &key_size, argv[0]);
+    if (!key)
+    {
+      return ccf::js::constants::Exception;
+    }
+
+    size_t unwrapping_key_size;
+    uint8_t* unwrapping_key =
+      JS_GetArrayBuffer(ctx, &unwrapping_key_size, argv[1]);
+    if (!unwrapping_key)
+    {
+      return ccf::js::constants::Exception;
+    }
+
+    js::Context& jsctx = *(js::Context*)JS_GetContextOpaque(ctx);
+
+    auto parameters = argv[2];
+    auto wrap_algo_name_val = jsctx(JS_GetPropertyStr(ctx, parameters, "name"));
+    JS_CHECK_EXC(wrap_algo_name_val);
+
+    auto wrap_algo_name_str = jsctx.to_str(wrap_algo_name_val);
+    if (!wrap_algo_name_str)
+    {
+      return ccf::js::constants::Exception;
+    }
+
+    try
+    {
+      auto algo_name = std::string(*wrap_algo_name_str);
```

Comment:
```suggestion
      auto algo_name = std::string(*wrap_algo_name_str);
```
wrap_algo_name_str is a std::optional<string>, auto algo_name = *wrap_algo_name_str is enough, the extra copy construction is unnecessary.